helpers/calendar/all_analytics_helper.py

```diff
@@ -0,0 +1,107 @@
+    bookings = graphene.Int()
+
+
+class AllAnalyticsHelper:
+
+    def bookings_count(self, unconverted_dates, events):
+        """
+        Get bookings count and period in a given room
+        """
+        start_date = unconverted_dates['start']
+        day_after_end_date = unconverted_dates['end']
+        bookings_count = []
+        start_dt = dateutil.parser.parse(start_date)
+        end_dt = dateutil.parser.parse(day_after_end_date)
+        number_of_days = (end_dt - start_dt).days
+
+        if number_of_days <= 30:
+            dates = CommonAnalytics.get_list_of_dates(
+                start_date, number_of_days)
+            for date in dates:
+                bookings = 0
+                string_date = dateutil.parser.parse(
+                    date[0]).strftime("%b %d %Y")
+                for event in events:
+                    if dateutil.parser.parse(event.start_time[:10]).strftime("%b %d %Y") == string_date: # noqa
+                        bookings += 1
+                output = BookingsCount(
+                    period=string_date,
+                    bookings=bookings)
+                bookings_count.append(output)
+        else:
+            dates = CommonAnalytics.get_list_of_month_dates(
+                start_date, start_dt, day_after_end_date, end_dt)
+            for date in dates:
+                string_month = dateutil.parser.parse(date[0]).strftime("%B")
+                bookings = 0
+                for event in events:
+                    if dateutil.parser.parse(event.start_time[:10]).strftime("%B") == string_month: # noqa
+                        bookings += 1
+                output = BookingsCount(
+                    period=string_month,
+                    bookings=bookings)
+                bookings_count.append(output)
+
+        return bookings_count
+
+    def get_all_analytics(self, query, start_date, end_date, location_id, unconverted_dates): # noqa
+        """
+        Get all room analytics
+        """
+        rooms = query.filter_by(state="active", location_id=location_id).all()
+        room_analytics = []
+        bookings = 0
+        for room in rooms:
+            cancellations = 0
+            checkins = 0
+            events = []
+            try:
+                # events_result - number of events in a given room
+                events_result = CommonAnalytics.get_all_events_in_a_room(
+                    self, room.id, start_date, end_date)
+            except GraphQLError:
+                continue
+            bookings_count = AllAnalyticsHelper.bookings_count(self, unconverted_dates, events_result) # noqa
+            bookings += len(events_result)
+            for event in events_result:
+                if event.checked_in:
+                    checkins += 1
+                if event.cancelled:
+                    cancellations += 1
+                duaration_in_minutes = CommonAnalytics.get_time_duration_for_event(self, event.start_time, event.end_time) # noqa
```

Comment:
put the parameters you are passing to the `get_time_duration_for_event` in individual lines and get rid of #noqa

Reply:
Fixed